lib/trino-collect/src/test/java/io/trino/collect/cache/TestEvictableCache.java

```diff
@@ -542,4 +546,42 @@ public void testInvalidateAndLoadConcurrently(Invalidation invalidation)
             executor.awaitTermination(10, SECONDS);
         }
     }
+
+    @Test(dataProvider = "disabledCacheImplementations")
+    public void testPutOnEmptyCache(DisabledCacheImplementation disabledCacheImplementation)
+    {
+        Map<Object, Object> cache = EvictableCacheBuilder.newBuilder()
+                .maximumSize(0)
+                .disabledCacheImplementation(disabledCacheImplementation)
+                .build()
+                .asMap();
+
+        int key = 0;
+        int value = 1;
+        assertThat(cache.put(key, value)).isNull();
+        assertThat(cache.put(key, value)).isNull();
+        assertThat(cache.putIfAbsent(key, value)).isNull();
+        assertThat(cache.putIfAbsent(key, value)).isNull();
+    }
+
+    @Test(dataProvider = "disabledCacheImplementations")
+    public void testPutOnNonEmptyCache(DisabledCacheImplementation disabledCacheImplementation)
+    {
+        Map<Object, Object> cache = EvictableCacheBuilder.newBuilder()
+                .maximumSize(10)
+                .disabledCacheImplementation(disabledCacheImplementation)
+                .build()
+                .asMap();
+
+        int key = 0;
+        int value = 0;
```

Comment:
use value=1, as above, so it's different from the key

Reply:
does it matter?

Reply:
@ssheikin  if it doesn't, why the difference?